client/services/FeedbackGeneratorService.js

```diff
@@ -222,6 +225,43 @@ tie.factory('FeedbackGeneratorService', [
         feedback.setSyntaxErrorIndex(feedback.getParagraphs().length - 1);
         return feedback;
       },
+      getPrereqFailureFeedback: function(codePrereqCheckResult) {
+        var feedback = FeedbackObjectFactory.create(false);
+        var preReqFailures = codePrereqCheckResult.getPrereqCheckFailures();
+
+        /* This function shouldn't be invoked unless there is at least one
+        pre-requisite check failure, but the code below is provided to prevent
+        an index out of bounds error. */
+        if (preReqFailures.length === 0) {
+          feedback.appendTextParagraph("Prerequisite checks passed!");
+          return feedback;
+        }
+
+        // Check first error type and generate appropriate feedback message.
+        var preReqFailure = preReqFailures[0];
+        if (preReqFailure.get('type') === PREREQ_CHECK_TYPE_MISSING_STARTER_CODE) {
```

Comment:
Added PrereqCheckFailureObjectFactory under client/domain with functions PrereqCheckFailure.isMissingStarterCode() and PrereqCheckFailure.isBadImport().